File: vel_pkg/src/vel_node.cpp
#include <rclcpp/rclcpp.hpp>
#include <geometry_msgs/msg/twist.hpp>

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);

    printf("Hello World, I am the vel node!\n");

    rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("vel_node");
    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr pub = node->create_publisher<geometry_msgs::msg::Twist>("/cmd_vel", 10);

    geometry_msgs::msg::Twist vel_msg;

    vel_msg.linear.x = 0.1;
    vel_msg.linear.y = 0.0;
    vel_msg.linear.z = 0.0;
    vel_msg.angular.x = 0.0;
    vel_msg.angular.y = 0.0;
    vel_msg.angular.z = 0.1;

    rclcpp::Rate loop_rate(30);

    while(rclcpp::ok())
    {
        pub->publish(vel_msg);

        loop_rate.sleep();
    }

    rclcpp::shutdown();
    return 0;
}

File: vel_pkg/vel_pkg/vel_node.py
#!/usr/bin/env python3
#conding: utf-8

import rclpy
from geometry_msgs.msg import Twist

def main(args=None):
    rclpy.init(args=None)

    node = rclpy.create_node('vel_node')

    print("Hello World, I am the aml node! [Python]\n")

    pub = node.create_publisher(Twist, '/cmd_vel', 10)

    rate = node.create_rate(30)  # 30 Hz

    while rclpy.ok():
        rclpy.spin_once(node)
        msg = Twist()

        msg.linear.x = 0.5
        msg.linear.y = 0.0
        msg.linear.z = 0.0

        msg.angular.x = 0.0
        msg.angular.y = 0.0
        msg.angular.z = 0.0

        pub.publish(msg)

        rate.sleep()

    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
